### ISSUE TITLE
FreeBSD build of 0.56 pollutes namespace with bogus `shift` function

### ISSUE THREAD

**Issue by greg-kennedy** [benkasminbullock/image-png-libpng#37]
This may be outside your view but I am having trouble with the latest FreeBSD port / package.  Latest version in `/usr/ports/graphics/p5-Image-PNG-Libpng` after running `make` produces an output Libpng.xs with this EXPORT_OK list:

```perl
our @EXPORT_OK = qw/
        access_version_number
        color_type_channels
        color_type_name
        copy_row_pointers
        create_read_struct
        ...
        set_unknown_chunks
        set_user_limits
        set_verbosity
        shift
        sig_cmp
        split_alpha
        text_compression_name
        write_end
        write_image
        write_info
        write_png
        write_to_scalar
        any2gray8
        copy_png
        create_reader
        create_writer
        get_internals
        image_data_diff
        png_compare
        read_png_file
        split_alpha
        write_png_file
/;
```

Looking in this list there is for some reason `shift`, which means that the built-in `shift` function is overwritten, and so this sample program:

```perl
use Image::PNG::Libpng ':all';

sub foo
{
  my $value = shift;
}

foo(1);
```

crashes with this error:
`Undefined subroutine &Image::PNG::Libpng::shift called at test.pl line 5.`

I'm not really familiar with the build process for this or how the @EXPORT_OK list is created, but, clearly `shift` should not be in there :)

**Comment by greg-kennedy** [comment]
See also similar bug in Debian for package "libpdf-builder-perl":
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=978331

**Comment by benkasminbullock** [comment]
It looks like there is a function png_set_shift and some commented-out code to support that accidentally got exported.

**Comment by benkasminbullock** [comment]
It's easy to remove the bad `shift`, but unfortunately another bug has appeared from somewhere, so releasing a new version depends on getting rid of the new bug. I'll post again here when it's ready to go.

**Comment by benkasminbullock** [comment]
That bug turned out to be very simple to fix, old libpngs seem to have required lots of dummy parameters, anyway I've put a release of the new version on CPAN:

https://metacpan.org/release/BKB/Image-PNG-Libpng-0.56_01

This has to go at least once past CPAN testers then I'll release 0.57.

I don't know when FreeBSD ports will change to that.

**Comment by greg-kennedy** [comment]
Thanks for the fix.  I'll nudge FreeBSD porters group when this is official.  In the meantime, a workaround is to not use ':all' and instead just the functions needed (e.g. 'create_write_struct')

**Comment by benkasminbullock** [comment]
I've released 0.57 to cpan just now:

https://metacpan.org/release/BKB/Image-PNG-Libpng-0.57

There's actually no test of the problem about shift, but I think it's resolved anyway.

I'd like to mark this as closed if that's OK with you.
